code/modules/mob/living/carbon/xenomorph/castes/warlock/abilities_warlock.dm

```diff
@@ -0,0 +1,416 @@
+// ***************************************
+// *********** Psychic shield
+// ***************************************
+/datum/action/xeno_action/activable/psychic_shield
+	name = "Psychic Shield"
+	ability_name = "psychic shield"
+	action_icon_state = "psy_shield"
+	mechanics_text = "Channel a psychic shield at your current location that can reflect most projectiles. Activate again while the shield is active to detonate the shield forcibly, producing knockback. Must remain static to use."
+	cooldown_timer = 8 SECONDS
+	plasma_cost = 200
+	keybinding_signals = list(
+		KEYBINDING_NORMAL = COMSIG_XENOABILITY_PSYCHIC_SHIELD,
+	)
+	keybind_flags = XACT_KEYBIND_USE_ABILITY
+	use_state_flags = XACT_USE_BUSY
+	var/obj/effect/xeno/shield/active_shield
+
+
+/datum/action/xeno_action/activable/psychic_shield/on_cooldown_finish()
+	to_chat(owner, span_notice("We gather enough mental strength to create a psychic shield again."))
+	return ..()
+
+/datum/action/xeno_action/activable/psychic_shield/use_ability(atom/A)
+	. = ..()
+	if(active_shield)
+		var/mob/living/carbon/xenomorph/xeno_owner = owner
+		if(plasma_cost > xeno_owner.plasma_stored)
+			to_chat(xeno_owner, span_warning("We need [plasma_cost - xeno_owner.plasma_stored] more plasma!"))
+			return FALSE
+		if(can_use_action(FALSE, XACT_USE_BUSY))
+			shield_blast()
+			cancel_shield()
+		return
+	var/turf/target_turf = get_step(owner, owner.dir)
+	if(target_turf.density)
+		owner.balloon_alert(owner, "Obstructed by [target_turf]")
+		return
+	for(var/atom/movable/affected in target_turf)
+		if(affected.density)
+			owner.balloon_alert(owner, "Obstructed by [affected]")
+			return
+
+	owner.visible_message(span_xenowarning("A strange psychic aura shimmers into life in front of \the [owner]!"), \
+	span_xenowarning("We form a barrier in front of us, we must hold still!"))
```

Comment:
does this visible message really need to exist since theres visuals and sound already

Reply:
Agree